app/controllers/users_email_controller.rb

```diff
@@ -36,12 +36,18 @@ def update
     RateLimiter.new(user, "change-email-min-#{request.remote_ip}", 3, 1.minute).performed!
 
     updater = EmailUpdater.new(guardian: guardian, user: user, initiating_user: current_user)
-    updater.change_to(params[:email])
+    updater.change_to(params[:email], add: request.post?)
 
     if updater.errors.present?
       return render_json_error(updater.errors.full_messages)
     end
 
+    if current_user.staff? && current_user != user
```

Comment:
Shouldn't this be done in the `EmailUpdater` class instead of the controller?

Also, we should not be logging the whole email address (cf. the `moderators_view_emails` site setting). I think just having a log that some staff member changed the email address of user X to be enough.